src/Response/Types/ErrorType.php

```diff
@@ -5,30 +5,166 @@
 namespace Core\Response\Types;
 
 use Core\Response\Context;
+use Core\Utils\CoreHelper;
+use Rs\Json\Pointer;
 
 class ErrorType
 {
     /**
      * Initializes a new object with the description and class name provided.
      */
-    public static function init(string $description, ?string $className = null): self
+    public static function init(string $description, ?string $className = null, bool $hasErrorTemplate = false): self
```

Comment:
Using any boolean flag is a code smell, create another public initailization method named `initWithTemplate` to avoid it